geometry/geometry/TrianglesView.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  View1.swift
+//  geometry
+//
+//  Created by Dmitry Tolmachev on 30.11.15.
+//  Copyright © 2015 Al Digit. All rights reserved.
+//
+
+import UIKit
+
+class TrianglesView: UIView {
+    private var dataSource: TrianglesProtocol?
+    
+    override func drawRect(rect: CGRect) {
+        UIColor.greenColor().set()
+        let context = UIGraphicsGetCurrentContext()
+        CGContextSetLineWidth(context, 3)
+        
+        if let trianglesDict = dataSource?.getTriangles(){
+            var n = 0
+            var numbers = [1:"1",2:"2",3:"3"]
+            
+            for n in 1...3 {
+            CGContextMoveToPoint(context, (trianglesDict[numbers[n]!]!["first"]!!["x"] as! CGFloat), (trianglesDict[numbers[n]!]!["first"]!!["y"] as! CGFloat))
```

Comment:
This has been refactored to a more shorter version when using arrays instead of dictionaries.